source/extensions/clusters/redis/redis_cluster.cc

```diff
@@ -0,0 +1,284 @@
+#include "redis_cluster.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Clusters {
+namespace Redis {
+
+RedisCluster::RedisCluster(
+    const envoy::api::v2::Cluster& cluster,
+    const envoy::config::cluster::redis::RedisClusterConfig& redisCluster,
+    NetworkFilters::Common::Redis::Client::ClientFactory& redis_client_factory,
+    Upstream::ClusterManager& clusterManager, Runtime::Loader& runtime,
+    Network::DnsResolverSharedPtr dns_resolver,
+    Server::Configuration::TransportSocketFactoryContext& factory_context,
+    Stats::ScopePtr&& stats_scope, bool added_via_api)
+    : Upstream::BaseDynamicClusterImpl(cluster, runtime, factory_context, std::move(stats_scope),
+                                       added_via_api),
+      cluster_manager_(clusterManager),
+      cluster_refresh_rate_(std::chrono::milliseconds(
+          PROTOBUF_GET_MS_OR_DEFAULT(redisCluster, cluster_refresh_rate, 5000))),
+      cluster_refresh_timeout_(std::chrono::milliseconds(
+          PROTOBUF_GET_MS_OR_DEFAULT(redisCluster, cluster_refresh_timeout, 3000))),
+      dispatcher_(factory_context.dispatcher()), dns_resolver_(std::move(dns_resolver)),
+      load_assignment_(cluster.has_load_assignment()
+                           ? cluster.load_assignment()
+                           : Config::Utility::translateClusterHosts(cluster.hosts())),
+      local_info_(factory_context.localInfo()), random_(factory_context.random()),
+      redis_discovery_session_(
+          std::make_unique<RedisDiscoverySession>(*this, redis_client_factory)) {
+  switch (cluster.dns_lookup_family()) {
```

Comment:
I'm guessing you copied this from somewhere. Can you move this into a common utility function somewhere?

Reply:
It appears this was indeed copied from somewhere. I'll put it in a utility.